test/jdk/java/net/Socket/UdpSocket.java

```diff
@@ -133,8 +132,17 @@ public void testMaxSockets() throws Exception {
         }
     }
 
+
     private Socket newUdpSocket() throws IOException {
-        return new Socket(InetAddress.getLoopbackAddress(), 8000, false);
+        Socket s = null;
+
+        try {
+            s = new Socket(InetAddress.getLoopbackAddress(), 8000, false);
+        } catch (BindException unexpected) {
```

Comment:
final BindException

Reply:
I used stdout for convenience as it aligns with the  two other output statements, so that I see immediately the retry output as follows:

----------System.out:(12/349)----------
[TestNG] Running:
  java/net/Socket/UdpSocket.java

BindException caught retry Socket creation
test UdpSocket.testMaxSockets(): success
test UdpSocket.testSendReceive(): success

===============================================

Reply:
I could have used return directly in multiple places ... but my style preference is a single exit point from a method